src/librustc/mir/mod.rs

```diff
@@ -3184,6 +3188,17 @@ impl<'tcx> TypeFoldable<'tcx> for PlaceBase<'tcx> {
     }
 }
 
+impl<'tcx> TypeFoldable<'tcx> for &'tcx ty::List<PlaceElem<'tcx>> {
```

Comment:
side-note: maybe some of the `MutVisitor`s should be `TypeFolder`s instead?

Reply:
Have you seen any in particular that you consider?

Reply:
all those that just apply an operation to all types.